baremetal/metal3machine_manager.go

```diff
@@ -390,6 +392,46 @@ func (m *MachineManager) Associate(ctx context.Context) error {
 		return err
 	}
 
+	if m.Metal3Machine.Spec.DataTemplate != nil {
+		// Requeue to get the DataTemplate output. We need to requeue to trigger the
+		// wait on the Metal3DataTemplate
+		if err := m.WaitForM3Metadata(ctx); err != nil {
+			return err
+		}
+
+		// If the requeue is not needed, then get the updated host and set the host
+		// specs
+		host, helper, err = m.getHost(ctx)
+		if err != nil {
+			m.SetError("Failed to get the BaremetalHost for the Metal3Machine",
+				capierrors.CreateMachineError,
+			)
+			return err
+		}
+
+		if err = m.setHostSpec(ctx, host); err != nil {
+			if _, ok := err.(HasRequeueAfterError); !ok {
+				m.SetError("Failed to set the BaremetalHost Specs",
+					capierrors.CreateMachineError,
```

Comment:
Yeah, agree, but at the same time if you check the manager code, you would find good number of `capierrors.CreateMachineError` calls in different use cases where we fail to get/set/associate/annotate certain objects, so that was the reason I followed the same pattern.